pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionCommon.java

```diff
@@ -108,17 +110,51 @@ public static int findAvailablePort() {
                 Type actualInputType = ((ParameterizedType) collectionType).getActualTypeArguments()[0];
                 typeArgs[0] = (Class<?>) actualInputType;
             }
+            if(typeArgs[0].equals(Message.class)) {
```

Comment:
whats the plan for adding this support to window function?

Reply:
we will probably have to change the interface from

```
public interface WindowFunction<I, O> {
    O process(Collection<Record<I>> input, WindowContext context) throws Exception;
}
```
to 

```
public interface WindowFunction<I, O> {
    O process(Collection<I> input, WindowContext context) throws Exception;
}
```

And support "I" as Record for BC and support "I" as Message.